src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-list/rgw-storage-class-list.component.ts

```diff
@@ -0,0 +1,100 @@
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+import { StorageClass, TierType } from '../models/rgw-storage-class.model';
+
+@Component({
+  selector: 'cd-rgw-storage-class-list',
+  templateUrl: './rgw-storage-class-list.component.html',
+  styleUrls: ['./rgw-storage-class-list.component.scss']
+})
+export class RgwStorageClassListComponent extends ListWithDetails implements OnInit {
+  columns: CdTableColumn[];
+  selection = new CdTableSelection();
+  tableActions: CdTableAction[];
+  storageClassList: StorageClass[] = [];
+
+  constructor(private rgwZonegroupService: RgwZonegroupService) {
+    super();
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`Storage Class`,
+        prop: 'storage_class',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Endpoint`,
+        prop: 'endpoint',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Region`,
+        prop: 'region',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Placement Target`,
+        prop: 'placement_target',
+        flexGrow: 2
+      },
+    ];
+  }
+
+  loadStorageClass(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.rgwZonegroupService.getAllZonegroupsInfo().subscribe(
+        (data: any) => {
+          this.storageClassList = [];
+          if (data && data?.zonegroups) {
+            data.zonegroups.forEach((zoneGroup: {
+              name: string; placement_targets: any[] }) => {
+              if (zoneGroup.placement_targets) {
+                zoneGroup.placement_targets.forEach((target) => {
+                  if (target.tier_targets) {
+                    target.tier_targets.forEach((tierTarget: any) => {
+                      if (tierTarget.val.tier_type === TierType) {
+                        const tierObj = {
+                          zonegroup_name: zoneGroup.name,
+                          placement_target: target.name,
+                          endpoint: tierTarget.val.s3.endpoint,
+                          access_key: tierTarget.val.s3.access_key,
+                          storage_class: tierTarget.val.storage_class,
+                          target_path: tierTarget.val.s3.target_path,
+                          region: tierTarget.val.s3.region,
+                          secret_key: tierTarget.val.s3.secret,
+                          multipart_min_part_size: tierTarget.val.s3.multipart_min_part_size,
+                          multipart_sync_threshold: tierTarget.val.s3.multipart_sync_threshold,
+                          host_style: tierTarget.val.s3.host_style
+                        };
+                        this.storageClassList.push(tierObj);
+                      }
+                    });
+                  }
+                });
+              }
+            });
+          }
+          resolve();
+        },
+        (error) => {
+          reject(error);
+        }
+      );
+    });
+  }
```

Comment:
We can use `async/await`  instead of `promises` here for better readability and maintenence.
Also define types :) , and consider using map, filter, or rxjs functions instead of multiple forEaches